src/main/java/org/minions/devfund/marylin/WhackAMole.java

```diff
@@ -0,0 +1,158 @@
+package org.minions.devfund.marylin;
+
+import java.util.Arrays;
+import java.util.Random;
+import java.util.Scanner;
+
+/**
+ * Class to simulate whack a mole.
+ */
+public class WhackAMole {
+    private int score;
+    private int molesLeft;
+    private int attemptsLeft;
+    private char[][] moleGrid;
+
+    /**
+     * Constructor of WhackAMole class.
+     *
+     * @param numAttempts   Number of the attempts.
+     * @param gridDimension Number of the grid dimension.
+     */
+    WhackAMole(int numAttempts, int gridDimension) {
+        this.moleGrid = new char[gridDimension][gridDimension];
+        this.attemptsLeft = numAttempts;
+
+        for (char[] row : this.moleGrid) {
+            Arrays.fill(row, '*');
+        }
+    }
+
+    /**
+     * Method to place a mole into the grid.
+     *
+     * @param x X axis in the grid.
+     * @param y Y axis in the grid.
+     * @return Returns true if the mole was place, false otherwise.
+     */
+    boolean place(int x, int y) {
+        if (this.moleGrid[x][y] != 'M') {
+            this.moleGrid[x][y] = 'M';
+            this.molesLeft++;
+            return true;
+        }
+        return false;
+    }
+
+    /**
+     * Method to whack a mole in the grid.
+     *
+     * @param x X axis in the grid.
+     * @param y Y axis in the grid.
+     */
+    void whack(int x, int y) {
+        if (this.moleGrid[x][y] == 'M') {
+            this.moleGrid[x][y] = 'W';
+            this.score++;
+            this.molesLeft--;
+        }
+        this.attemptsLeft--;
+    }
+
+    /**
+     * Method to print the status of the game without shows the moles.
+     */
+    void printGridToUser() {
+        for (int i = 0; i < this.moleGrid.length; i++) {
+            for (int j = 0; j < this.moleGrid.length; j++) {
+                if (this.moleGrid[i][j] == 'M') {
+                    System.out.print("*  ");
+                } else {
+                    System.out.print(this.moleGrid[i][j] + "  ");
+                }
+            }
+            System.out.println("");
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Replace this use of System.out or System.err by a logger. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS106)